rasterio/_io.pyx

```diff
@@ -865,9 +865,9 @@ cdef class RasterReader(_base.DatasetReader):
                 roff = 0
                 coff = 0
                 if window[0][0] < 0:
-                    roff = int(round(window_h*scaling_h)) - data_h
+                    roff = (0 - window[0][0]) * scaling_h
                 if window[1][0] < 0:
-                    coff = int(round(window_w*scaling_w)) - data_w
+                    coff = (0 - window[1][0]) * scaling_w
```

Comment:
No reason, i just arrived at the solution through additive math and left it at that. negation is more clear though.